File: commands/utility/serverinfo.go
package utility

import (
  "github.com/sapphire-cord/sapphire"
  "fmt"
  "github.com/bwmarrin/discordgo"
  "strconv"
)

// Gets information about the current server.
// Aliases: si, gi, guildinfo, guild, server
// guild only
func Serverinfo(ctx *sapphire.CommandContext) {
  ownerMember, err := ctx.Session.State.Member(ctx.Guild.ID, ctx.Guild.OwnerID)
  if err != nil {
    ctx.Reply("Failed to fetch owner details: %s", err)
    return
  }
  owner := ownerMember.User
  ctx.BuildEmbed(sapphire.NewEmbed().
    SetTitle("Server Info.").
    SetDescription(fmt.Sprintf("%s (%s)", ctx.Guild.Name, ctx.Guild.ID)).
    AddField("Members", strconv.Itoa(ctx.Guild.MemberCount)).
    AddField("Roles", strconv.Itoa(len(ctx.Guild.Roles))).
    AddField("Emojis", strconv.Itoa(len(ctx.Guild.Emojis))).
    AddField("Channels", strconv.Itoa(len(ctx.Guild.Channels))).
    AddField("Owner", fmt.Sprintf("%s#%s (%s)", owner.Username, owner.Discriminator, owner.ID)).
    SetColor(0xDFAC7C).
    SetAuthor(ctx.Author.Username, ctx.Author.AvatarURL("256")).
    SetThumbnail(discordgo.EndpointGuildIcon(ctx.Guild.ID, ctx.Guild.Icon)))
}

File: commands/utility/serverinfo.js
const Command = require("../../structures/Command.js");
const { MessageEmbed } = require("discord.js");

class ServerInfo extends Command {
  constructor(...args) {
    super(...args, {
      guildOnly: true,
      aliases: ["guild", "si", "server"],
      description: "Get information on the current server."
    });
    
    this.verificationLevels = {
      "NONE": "None",
      "LOW": "Low",
      "MEDIUM": "Medium",
      "HIGH": "(╯°□°）╯︵ ┻━┻",
      "VERY_HIGH": "┻━┻ ﾐヽ(ಠ益ಠ)ノ彡┻━┻"
    };
    
    this.filterLevels = {
      "DISABLED": "Off",
      "MEMBERS_WITHOUT_ROLES": "No Role",
      "ALL_MEMBERS": "Everyone"
    };
  }

  async run(ctx) {
    const days = Math.floor((new Date() - ctx.guild.createdAt) / (1000 * 60 * 60 * 24));
    const bans = await ctx.guild.fetchBans()
      .then((bans) => bans.size)
      .catch(() =>  "Couldn't fetch bans.");

    if(!ctx.guild.owner) await ctx.guild.members.fetch(ctx.guild.ownerID).catch(() => null);

    const embed = new MessageEmbed()
      .setColor(0x9590EE)
      .setThumbnail(ctx.guild.iconURL())
      .addField("❯ Name", ctx.guild.name, true)
      .addField("❯ ID", ctx.guild.id, true)
      .addField("❯ Creation Date", `${ctx.guild.createdAt.toDateString()} (${days} days ago!)`, true)
      .addField("❯ Region", ctx.guild.region, true)
      .addField("❯ Explicit Filter", this.filterLevels[ctx.guild.explicitContentFilter], true)
      .addField("❯ Verification Level", this.verificationLevels[ctx.guild.verificationLevel], true)
      .addField("❯ Owner", ctx.guild.owner ? ctx.guild.owner.user.tag : "Failed to get owner information.", true)
      .addField("❯ Members", `${ctx.guild.memberCount}`, true)
      .addField("❯ Ban Count", bans, true);
    return ctx.reply({ embed });
  }
}

module.exports = ServerInfo;
